src/Tizen.Wearable.CircularUI.Chart.Forms.Renderer/BarChartViewRenderer.cs

```diff
@@ -0,0 +1,752 @@
+                DrawMajorAxisLine(canvas);
+            }
+
+            if (Element.AxisOption.IsVisibleOfCategoryLabel)
+            {
+                DrawCategoryLabels(canvas, points);
+            }
+
+            if (Element.AxisOption.IsVisibleOfMinorAxisLine)
+            {
+                DrawMinorAxisLine(canvas);
+            }
+
+            if (Element.AxisOption.IsVisibleOfReferenceLabel)
+            {
+                DrawReferenceLabels(canvas, barSize);
+            }
+
+            if (Element.BarBackgroundColorIsVisible)
+            {
+                DrawBarBackground(canvas, points, barSize);
+            }
+
+            if (Element.AxisOption.IsVisibleOfReferenceLine)
+            {
+                DrawReferenceLines(canvas, barSize);
+            }
+
+            DrawBars(canvas, points, barSize);
+
+            if (Element.ValueLabelIsVisible)
+            {
+                DrawValueLabels(canvas, points, barSize);
+            }
+        }
+
+        protected virtual void GenerateDataTable()
+        {
+            if (Element.Data == _prevDataSetArray)
+                return;
+
+            _barChartDataTable = new DataItem[_categoryCount];
+            var items = Element.Data[0].DataItems;
+            for (int i = 0; i < items.Count; i++)
+            {
+                var index = items[i].Key == 0 ? i : items[i].Key - 1;
+                index = index < _categoryCount ? index : _categoryCount - 1;
+                _barChartDataTable[index] = items[i] as DataItem;
+            }
+
+            _prevDataSetArray = Element.Data;
+        }
+
+        protected void CaculateAxisSize(SKCanvas canvas)
+        {
+            if (Element.AxisOption == null)
+                return;
+
+            if (Element.AxisOption.IsVisibleOfMajorAxisLine)
+            {
+                _majorAxisSize.Height = LineSize;
+            }
+
+            if (Element.AxisOption.IsVisibleOfMinorAxisLine)
+            {
+                _minorAxisSize.Width = LineSize;
+            }
+
+            if (Element.AxisOption.IsVisibleOfCategoryLabel)
+            {
+                CalculateCategoryItemSize();
+            }
+
+            if (Element.AxisOption.IsVisibleOfReferenceLabel)
+            {
+                CalculateReferenceItemSize();
+            }
+        }
+
+        private SKRect[] MeasureCategoryLabels()
+        {
+            using (var paint = new SKPaint())
+            {
+                return Element.AxisOption.CategoryLabels.Select(e =>
+                {
+                    if (string.IsNullOrEmpty(e.Label.Text))
+                    {
+                        return SKRect.Empty;
+                    }
+
+                    var bounds = new SKRect();
+                    var text = e.Label.Text;
+                    paint.TextSize = (float)(e.Label.FontSize * FontCovertConst);
+                    paint.MeasureText(text, ref bounds);
+                    return bounds;
+                }).ToArray();
```

Comment:
When I removed ToArray(), It makes crash when entering test app. So I remain this code.

Reply:
Could you tell me which exception was thrown?

Reply:
CallStack  doesn't provide in sdb dlog. and also I can't extract log file from crash dump file because file always is damaged with this crash. you can easily reproduce this issue with CircularUIChartGallery app with removing ToArray().

Reply:
Done. moved using( var paint .. ) code within select code.